Frameworks/UIKit/NSString+UIKitAdditions.mm

```diff
@@ -259,9 +259,8 @@ - (CGSize)_sizeWithAttributes:(NSDictionary<NSString*, id>*)attributes constrain
         size.width = std::numeric_limits<CGFloat>::max();
     }
 
-    if (size.height == 0.0) {
-        size.height = std::numeric_limits<CGFloat>::max();
-    }
+    // Return the height that the text actually requires rather than the constrained size
+    size.height = std::numeric_limits<CGFloat>::max();
```

Comment:
so wait the expectation of the constrainedToSize: contract is that the height passed in doesn't matter at all? that seems... wrong...